File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java
import java.util.Arrays;

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    private static final int[] lastWeekCounts = new int[] {0, 2, 5, 3, 7, 8, 4 };

    public int[] getLastWeek() {
        return lastWeekCounts;
    }

    public int getToday() {
        return birdsPerDay[birdsPerDay.length - 1];
    }

    public void incrementTodaysCount() {
        int lastIndex = birdsPerDay.length - 1;
        birdsPerDay[lastIndex]++;
    }

    public boolean hasDayWithoutBirds() {
        return Arrays.stream(birdsPerDay)
                     .anyMatch(x -> x == 0);
    }

    public int getCountForFirstDays(int numberOfDays) {
        return Arrays.stream(birdsPerDay)
                     .limit(numberOfDays)
                     .sum();
    }

    public int getBusyDays() {
        return (int) Arrays.stream(birdsPerDay)
                           .filter(x -> x >= 5)
                           .count();
    }
}

File: solutions/csharp/bird-watcher/1/BirdWatcher.cs
using System;
using System.Linq;

internal class BirdCount
{
	private readonly int[] _birdsPerDay;

	public BirdCount(int[] birdsPerDay)
	{
		_birdsPerDay = birdsPerDay;
	}

	private static readonly int[] _lastWeekCounts = new int[] { 0, 2, 5, 3, 7, 8, 4 };

	public static int[] LastWeek()
	{
		return _lastWeekCounts;
	}

	public int Today()
	{
		return _birdsPerDay.Last();
	}

	public void IncrementTodaysCount()
	{
		int lastIndex = _birdsPerDay.Length - 1;
		_birdsPerDay[lastIndex]++;
	}

	public bool HasDayWithoutBirds()
	{
		return _birdsPerDay.Contains(0);
	}

	public int CountForFirstDays(int numberOfDays)
	{
		return _birdsPerDay[..numberOfDays]
			.Sum();
	}

	public int BusyDays()
	{
		return _birdsPerDay
			.Where(x => x >= 5)
			.Count();
	}
}
